### DISCUSSION TITLE
Removal of the auto-paste commands

### DISCUSSION THREAD

**Discussion by RyotaUshio** [RyotaUshio/obsidian-pdf-plus#112] [Announcements]
The **Copy & auto-paste link to selection or annotation** command is planned to be removed in the near future. Instead, use the **Copy link to selection or annotation** command after enabling auto-paste using the **Toggle auto-paste** icon in the left ribbon menu or the **Toggle auto-paste** command.

A benefit of separating the copy command and the auto-paste mechanism is that you can combine any actions to copy PDF links (not just the command) with auto-paste (just like auto-focus). For example: https://github.com/RyotaUshio/obsidian-pdf-plus/issues/102#issuecomment-2001973233

If you don't like this change, feel free to share your thoughts in this thread.

**Comment by N3C2L** [comment]
My current workflow involves executing my most frequently used custom copy format with the command "Copy & auto-paste link to selection or annotation" after highlighting the text, using the keyboard shortcut bound to the NumPad 1 key on my keyboard.

Does this mean that in the future, this command will also execute when "toggle auto-paste" is enabled, and afterward "Copy link to selection or annotation" is executed? Because I just tested this an it does not auto-paste.

I would like to be able to execute other link formats with additional hotkeys. For example, I would use NumPad 2 to auto-paste a link with the copy format for research, and NumPad 3 to auto-paste a link with another format etc. Instead of right clicking every time after highlighting the text, I would then just press Numpad 1,2 or 3. Perhaps something like this would be possible if the mechanism for copy and auto-paste were separated?

**Reply by RyotaUshio** [reply]
> Does this mean that in the future, this command will also execute when "toggle auto-paste" is enabled, and afterward "Copy link to selection or annotation" is executed? Because I just tested this an it does not auto-paste.

Sorry I'm not sure if I understand this part correctly, but what I mean is that, when auto-paste is enabled, the "Copy link to selection or annotation" command does exactly the same thing as "Copy **& auto-paste** link to selection or annotation".
For example,
1. Assign a hotkey to "Copy link to selection or annotation" (not the auto-paste command)
2. Enable auto-paste by clicking the "Toggle auto-paste" icon in the left ribbon
3. Select some text and press that hotkey

https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/733fc3e1-2e04-4df7-ab46-e60df2f951c5

When auto-paste is enabled, Auto-pasting is also triggered when clicking a color palette or copying a link from the context menu.

For example: 

https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/0540d81f-30eb-4055-b3c6-0d5156fc483b

> I would like to be able to execute other link formats with additional hotkeys. 

This would be a nice feature and I'd like to implement it but there are some tricky parts so it will take some time until it comes true.

**Reply by N3C2L** [reply]
I am sorry. I did something wrong. Yes it works everything you mentioned above.
I was doing a mistake. My mistake was, that after highlighting the text, I executed the command via the command palette and that didn't work. The command was showing up because I am using the plugin "better command palette". But in the native command palette of obsidian this command wouldn't have appeared here, correctly.

**Reply by RyotaUshio** [reply]
I see, no need for apologies! I have to write a documentation page for this shortly...

> Perhaps something like this would be possible if the mechanism for copy and auto-paste were separated?

yes, it will be much easier with copy and auto-pasted separated.

So are you positive about this change?

**Reply by N3C2L** [reply]
Yes, I am positive about this change & thanks for your documentations. They are indeed useful for me & everyone else.

**Comment by Elberto123** [comment]
Hello,
I was able to use this solution and it worked.

1. Assign a hotkey to "Copy link to selection or annotation" (not the auto-paste command)
2. Enable auto-paste by clicking the "Toggle auto-paste" icon in the left ribbon
3. Select some text and press that hotkey...

However, it doesn't copy the annotations from the PDF, only the highlighted. Is there something I should activate or deactivate to solve this?

**Reply by RyotaUshio** [reply]
> it doesn't copy the annotations from the PDF, only the highlighted

I'm not sure what you meant here. Are you trying to copy a link to an annotation that is written directly in the PDF?
If you are doing so by re-selecting the range of text that is annotated and pressing the hotkey, it's not how it works.

Let me know if it's not what you are doing - some screen recording will make it easier to understand the situation.

**Reply by Elberto123** [reply]
I apologize, but I don't speak English. I'll try to be clearer.

The underlined part transfers, but the comment inserted within the underlined part does not.

Is it possible to extract the comment or was it never possible?


https://github.com/RyotaUshio/obsidian-pdf-plus/assets/165972273/12f0f2df-7c30-477e-b382-e563d808b5d7

**Reply by N3C2L** [reply]
@Elberto123 

Maybe you could comment directly in the .md annotation file instead inside the PDF file?
Because in your video you did the same what @RyotaUshio described in his comment above: 

> If you are doing so by re-selecting the range of text that is annotated and pressing the hotkey, it's not how it works.

**Reply by samanthavenialogan** [reply]
@N3C2L  pointed over to this thread as it more directly approaches the problem that I am having, but I already submitted a reply on my original link available here: 

https://github.com/RyotaUshio/obsidian-pdf-plus/discussions/132#discussioncomment-9440872

Suffice to say here, WE NEED to be able to pull annotations in addition to hilights and quotes.  Ithink it's already even available for that change if we can just change where the 'text' variable points to since we already have synonyms for that attribute (both text and citation pull the same data at the moment so just switch one to pull the annotation instead). 

As I understand it attribute availability is controlled by obsidian-core though, and not this plugin? 

**My comment over there for ease of reading:**
![obsidian feature request](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/54729415/1ba55a86-db7b-450b-b76f-c17ea3510db5)
